Sources/Apexy/Endpoint.swift

```diff
@@ -14,6 +14,9 @@ public protocol Endpoint {
     ///
     /// - Author: Nino
     associatedtype Content
+    
+    /// Error type
+    associatedtype ErrorType: Error
```

Comment:
Swift `Result` and `Combine.Publisher` have a related value called `Failure`. It would be nice to do like them.